b2sdk/transfer/emerge/emerger.py

```diff
@@ -63,6 +63,12 @@ def emerge(
         """
         planner = self.get_emerge_planner(recommended_upload_part_size=recommended_upload_part_size)
         emerge_plan = planner.get_emerge_plan(write_intents)
+        if emerge_plan.is_large_file():
+            [write_intent] = write_intents
+            if write_intent.outbound_source.is_sha1_known():
```

Comment:
If you want to write it right you would need to check if all `emerge_plan.emerge_parts` `.part_definition` are instances of `UploadEmergePartDefinition` **and** if `.part_definition.upload_source` is the same object in every part - if the answer is yes - then you can set `large_file_sha1` is it is known for this **common** `upload_source`